build/Dockerfile

```diff
@@ -1,9 +1,6 @@
 FROM --platform=${BUILDPLATFORM} golang:1.23 AS go_builder
 WORKDIR /go/src/github.com/percona/percona-server-mongodb-operator
 
-COPY go.mod go.sum ./
-RUN go mod download
-
```

Comment:
why do we need this change? with this we'll need to download modules after a every code change

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1826/commits/ef28b079a6966c9ddb767cd65b931fedd9eec621